src/shogun/machine/Machine.h

```diff
@@ -255,6 +274,21 @@ class Machine : public StoppableSGObject
 			return false;
 		}
 
+		virtual std::shared_ptr<Machine> fit_machine(
+		    const std::shared_ptr<Features>& data,
+		    const std::shared_ptr<Labels>& lab)
+		{
+			error("train_machine is not yet implemented for {}!", get_name());
+			return nullptr;
+		}
+
+		virtual std::shared_ptr<Labels>
+		predict_labels(const std::shared_ptr<Features>& data) const
+		{
+			error("train_machine is not yet implemented for {}!", get_name());
+			return nullptr;
+		}
```

Comment:
same comment as in `fit_machine`

Reply:
These error msgs should be different. A SG_NOTIMPLEMENTED exception would be appripriate